server/conn_stmt_test.go

```diff
@@ -399,93 +312,6 @@ func TestCursorExistsFlag(t *testing.T) {
 	require.False(t, mysql.HasCursorExistsFlag(getLastStatus()))
 }
 
-func TestCursorReadWithRCCheckTS(t *testing.T) {
```

Comment:
I remember there are quite some issues with the cursor fetch mode, for example, the `statement context` would be reset processing the next `fetch` command and the information in it could be lost processing.
For this test case I think it's ok to fetch all the results at the same time, the thing I'm worried about is that if there would be a memory problem if all the results are stored on the tidb-server side, usually when the cursor mode is used the result set would be large, it may increase the OOM risk.

Reply:
> I remember there are quite some issues with the cursor fetch mode, for example, the statement context would be reset processing the next fetch command and the information in it could be lost.

Yes. This PR is trying to solve this problem :thinking:. These context are not needed in the `FETCH` command now.

> I'm worried about is that there would be a memory problem if all the results are stored on the tidb-server side, usually when the cursor mode is used the result set would be large, it may increase the OOM risk.

Can't agree more :crying_cat_face: